onadata/apps/restservice/services/kpi_hook.py

```diff
@@ -19,13 +19,15 @@ def send(self, endpoint, data):
         }
         headers = {"Content-Type": "application/json"}
 
-        # Build the url in the service to avoid saving hardcoded domain name in the DB
-        url = "{}{}".format(
-            settings.KPI_INTERNAL_URL,
-            endpoint
-        )
-        response = requests.post(url, headers=headers, json=post_data)
-        response.raise_for_status()
+        # Verify if endpoint starts with `/assets/` before sending the request to`kpi`
+        if endpoint.startswith(settings.KPI_HOOK_ENDPOINT_PATTERN[:8]):
+            # Build the url in the service to avoid saving hardcoded domain name in the DB
+            url = "{}{}".format(
+                settings.KPI_INTERNAL_URL,
+                endpoint
+            )
+            response = requests.post(url, headers=headers, json=post_data)
+            response.raise_for_status()
 
-        # Save successful
-        Instance.objects.filter(pk=data.get("instance_id")).update(posted_to_kpi=True)
+            # Save successful
+            Instance.objects.filter(pk=data.get("instance_id")).update(posted_to_kpi=True)
```

Comment:
I want to nudge you on this pattern a little bit. If there's a block of existing code like:
```python
def lovely():
    here_i_am = doing_great_things()
    all_the_greatest = []
    for favorite in all_the_treats():
        if favorite.really_is_the_best():
            all_the_greatest.append(favorite)
    return all_the_greatest
```
…and we realize later that this code should only run when a certain condition is met, your preference seems to be enclosing the whole block in an `if`:
```python
def lovely():
    if not the_sky_is_falling:
        here_i_am = doing_great_things()
        all_the_greatest = []
        for favorite in all_the_treats():
            if favorite.really_is_the_best():
                all_the_greatest.append(favorite)
        return all_the_greatest
```
I don't like this :grin: I would rather handle the situation that precludes normal execution in its own conditional block, without enclosing / indenting the existing code:
```python
def lovely():
    if the_sky_is_falling:
        return None # or log, or raise an exception!
    here_i_am = doing_great_things()
    all_the_greatest = []
    for favorite in all_the_treats():
        if favorite.really_is_the_best():
            all_the_greatest.append(favorite)
    return all_the_greatest
````
As a reader, this lets my little brain say "phew!" after seeing the `if` block start and end within a few lines. Longer blocks have me thinking, "wait, where was I again?", re-reading, scrolling, etc. Also, leaving the indentation of the existing code alone leads to a smaller diff, which is easier to review.

To be fair, this is not a particularly tough example, but I hope my point comes across.

Reply:
We've discussed about this in on another PR.
The code will be changed accordingly.